File: apps/gig-crawler-2/src/models/venue.ts
import { z } from "zod";

export const VenueSchema = z.object({
  name: z.string(),
  address: z.string().optional(),
  website: z.string().optional(),
});

export type Venue = z.infer<typeof VenueSchema>;

export const StrapiVenueSchema = z.object({
  data: z.object({
    name: z.string(),
    address: z.string().optional(),
    website: z.string().optional(),
  }),
});

export type StrapiVenue = z.infer<typeof StrapiVenueSchema>;

export function toStrapiVenue(venue: Venue): StrapiVenue {
  return {
    data: {
      name: venue.name,
      address: venue.address,
      website: venue.website,
    },
  };
}

File: apps/gig-crawler-2-python-backup/src/models/venue.py
"""Venue models matching Strapi schema."""
from typing import Optional
from pydantic import BaseModel, Field


class Venue(BaseModel):
    """Venue model for business logic."""
    name: str
    address: Optional[str] = None
    website: Optional[str] = None


class StrapiVenue(BaseModel):
    """Venue model for Strapi API."""
    data: dict = Field(...)

    @classmethod
    def from_venue(cls, venue: Venue) -> "StrapiVenue":
        """Convert Venue to Strapi format."""
        return cls(
            data={
                "name": venue.name,
                "address": venue.address,
                "website": venue.website,
            }
        )
